sdk/include/platform/sunburst/platform-gpio.hh

```diff
@@ -127,6 +279,101 @@ struct SonataGPIO
 	 */
 	SonataJoystick read_joystick() volatile
 	{
-		return static_cast<SonataJoystick>(input & 0x1f);
+		return static_cast<SonataJoystick>(input & Inputs::Joystick);
 	}
 };
+
+/**
+ * A driver for Sonata's Raspberry Pi HAT Header GPIO.
+ *
+ * Documentation source:
+ * https://lowrisc.org/sonata-system/doc/ip/gpio.html
+ */
+struct SonataGpioRaspberryPiHat
+  : Internal::SonataGpioBase<SonataGpioRaspberryPiHat>
+{
+	/**
+	 * The mask of bits of the `output` register that contain meaningful GPIO
+	 * output.
+	 */
+	static const uint32_t OutputMask = 0x0FFF'FFFF;
+
+	/**
+	 * The mask of bits of the `input` and `debouncedInput` registers that
+	 * contain meaningful GPIO input.
+	 */
+	static const uint32_t InputMask = 0x0FFF'FFFF;
+
+	/**
+	 * The mask of bits of the `outputEnable` register that correspond to GPIO
+	 * pins which can have their output enabled or disabled.
+	 */
+	static const uint32_t OutputEnableMask = 0x0FFF'FFFF;
+};
+
+/**
+ * A driver for Sonata's Arduino Shield Header GPIO.
+ *
+ * Documentation source:
+ * https://lowrisc.org/sonata-system/doc/ip/gpio.html
+ */
+struct SonataGpioArduinoShield
+  : Internal::SonataGpioBase<SonataGpioArduinoShield>
+{
+	/**
+	 * The mask of bits of the `output` register that contain meaningful GPIO
+	 * output.
+	 */
+	static const uint32_t OutputMask = 0x0003'FFFF;
+
+	/**
+	 * The mask of bits of the `input` and `debouncedInput` registers that
+	 * contain meaningful GPIO input.
+	 */
+	static const uint32_t InputMask = 0x0003'FFFF;
+
+	/**
+	 * The mask of bits of the `outputEnable` register that correspond to GPIO
+	 * pins which can have their output enabled or disabled.
+	 */
+	static const uint32_t OutputEnableMask = 0x0003'FFFF;
+};
+
+/**
+ * A driver for Sonata's PMOD Header GPIO.
+ *
+ * Documentation source:
+ * https://lowrisc.org/sonata-system/doc/ip/gpio.html
+ */
+struct SonataGpioPmod : Internal::SonataGpioBase<SonataGpioPmod>
+{
+	/**
+	 * The mask of bits of the `output` register that contain meaningful GPIO
+	 * output.
+	 */
+	static const uint32_t OutputMask = 0x0000'FFFF;
+
+	/**
+	 * The mask of bits of the `input` and `debouncedInput` registers that
+	 * contain meaningful GPIO input.
+	 */
+	static const uint32_t InputMask = 0x0000'FFFF;
+
+	/**
+	 * The mask of bits of the `outputEnable` register that correspond to GPIO
+	 * pins which can have their output enabled or disabled.
+	 */
+	static const uint32_t OutputEnableMask = 0x0000'FFFF;
+
+	/**
+	 * The bit mappings of the two PMOD headers available in Sonata's PMOD GPIO.
+	 */
+	enum Headers : uint32_t
+	{
+		Pmod0 = (0xFF << 0),
+		Pmod1 = (0xFF << 8),
+	};
+};
+
+// General GPIO typedef for backwards compatibility
+typedef SonataGpioGeneral SonataGPIO;
```

Comment:
Have [a depreciated attribute](https://clang.llvm.org/docs/AttributeReference.html#deprecated) here would be dope